PHOENIX-6692 Add HBase 2.5 support

stoty opened the pull request.

stoty commented:
We have test failures, need too look into them.
[ERROR] Errors:
[ERROR]   MaxLookbackIT.testRecentMaxVersionsNotCompactedAway:315->majorCompact:343 » TestTimedOut
[ERROR]   MaxLookbackIT.testRecentlyDeletedRowsNotCompactedAway:154->majorCompact:343 » TestTimedOut
[ERROR]   MaxLookbackIT.testTTLAndMaxLookbackAge:238->majorCompact:343 » TestTimedOut te...
[ERROR]   TransformMonitorExtendedIT.testTransformIndexWithNamespaceEnabled:123 » SQL ER...
[ERROR]   TransformMonitorExtendedIT.testTransformTableWithNamespaceEnabled:155 » SQL ER...
[ERROR]   TransformMonitorExtendedIT.testTransformWithGlobalAndTenantViews:246 » SQL ERR...

stoty commented:
This is not expected to compile on Jenkins until official 2.5.0 artifacts are available.
While we recompile 2.4 with Hadoop3 to run the tests, the other compatibility modules use the public artifacts for compilation, and will fail.

For compiling the compatibility modules Hadoop2/Hadoop3 is not an issue, the Hadoop3 compiled HBase is only needed for the shaded JARs and the minicluster tests.

richardantal reviewed (commented):
Looks good to me
I built HBase 2.5.0 locally and than ran `mvn clean package -DskipTests -Dhbase.profile=2.5` successfully.

Locally I had some test failures but It fails with the default 2.4 HBase profiles as well, probably something is not OK on my side.

stoty commented on phoenix-hbase-compat-2.5.0/pom.xml:
```diff
@@ -0,0 +1,113 @@
+    </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
```
That version is defined in the main pom.

Also, even though 2.5.0 hasn't been released yet, branch-2.5 identified itself as 2.5.1. I expect that it's going to be reset
before release, but for now you need to override the version while building:

`mvn clean package -Dhbase.profile=2.5 -Dhbase.version=2.5.1-SNAPSHOT`

richardantal replied:
I used the 2.5.0RC0 branch/tag which had 2.5.0 version.

stoty commented:
2.5.0 is out.
I have updated the CI setup to also build for 2.5, and also added 5.2.0 to the upgrade tests for Hbase 2.3 and 2.4.

I think we can commit this now. (Once the tests have run)

stoty commented:
PTAL @gjacoby126 @richardantal

stoty commented:
Fixed some checkstlye issues.

gjacoby126 reviewed (approved):
+1, thanks @stoty
